src/gui/log/loglistview.cpp

```diff
@@ -0,0 +1,124 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "loglistview.h"
+
+#include <QApplication>
+#include <QClipboard>
+#include <QFontMetrics>
+#include <QKeyEvent>
+#include <QPainter>
+#include <QStyle>
+#include <QStyledItemDelegate>
+
+#include "logmodel.h"
+#include "uithememanager.h"
+
+namespace
+{
+    const QString SEPERATOR = QStringLiteral(" - ");
+
+    class LogItemDelegate : public QStyledItemDelegate
+    {
+    public:
+        using QStyledItemDelegate::QStyledItemDelegate;
+
+    private:
+        void paint(QPainter *painter, const QStyleOptionViewItem &option, const QModelIndex &index) const override
+        {
+            const QPen originalPen = painter->pen();
+            painter->save();
+
+            QStyledItemDelegate::paint(painter, option, index); // paints background, focus rect and selection rect
+
+            const QStyle *style = option.widget ? option.widget->style() : QApplication::style();
+            const int textHMargin = style->pixelMetric(QStyle::PM_FocusFrameHMargin, &option, option.widget) + 1;
+            const int textVMargin = style->pixelMetric(QStyle::PM_FocusFrameVMargin, &option, option.widget);
+            const QRect textRect = style->subElementRect(QStyle::SE_ItemViewItemText, &option, option.widget)
+                                   .adjusted(textHMargin, textVMargin, -textHMargin, -textVMargin);
+
+            QPen coloredPen = originalPen;
+            coloredPen.setColor(Qt::gray);
+            painter->setPen(coloredPen);
+            const QString time = index.data(BaseLogModel::TimeRole).toString();
+            style->drawItemText(painter, textRect, 0, option.palette, (option.state & QStyle::State_Enabled), time);
+
+            painter->setPen(originalPen);
+            const int timeWidth = option.fontMetrics.width(time);
+            style->drawItemText(painter, textRect.adjusted(timeWidth, 0, 0, 0), 0, option.palette, (option.state & QStyle::State_Enabled), SEPERATOR);
+
+            const QVariant foreground = index.data(Qt::ForegroundRole);
+            if (foreground.canConvert<QColor>()) { // peermodel may give invalid foreground role
```

Comment:
>// peermodel may give invalid foreground role

Is it intentional behavior? What reason it does for?

Reply:
for the peer model we paint with default font color, I already changed the pen to `originalPen` for the painting of Separator so when this condition fails the following message will be painted with the default font color.

I think I should change it to 
// peermodel **will** give invalid foreground role

Reply:
ok it seems originally we paint peer messages with red color will change it accordingly